To: Greg Kroah-Hartman <gregkh@xxxxxxxxxxxxxxxxxxx>
Subject: [PATCH v1] usb: xhci: plat: remove error log for failure to get usb-phy
From: Stanley Chang <stanley_chang@xxxxxxxxxxx>
Date: Wed, 10 May 2023 15:51:29 +0800
Cc: Stanley Chang <stanley_chang@xxxxxxxxxxx>, Klaus Kudielka <klaus.kudielka@xxxxxxxxx>, Mathias Nyman <mathias.nyman@xxxxxxxxx>, <linux-usb@xxxxxxxxxxxxxxx>, <linux-kernel@xxxxxxxxxxxxxxx>
Remove this log to avoid non-error conditions.
If CONFIG_USB_PHY is disabled, the following error message appears:
[    0.231609] xhci-hcd f10f0000.usb3: xhci_plat_probe get usb3phy fail (ret=-6)
[    0.239716] xhci-hcd f10f8000.usb3: xhci_plat_probe get usb3phy fail (ret=-6)
In this case, devm_usb_get_phy_by_phandle is declared static inline
and returns -ENXIO.

It is easy to pinpoint the failure to get the usb-phy using the debug
log in drivers/usb/phy/phy.c. Therefore, it can be removed.

https://www.spinics.net/lists/kernel/msg4783770.html

Signed-off-by: Stanley Chang <stanley_chang@xxxxxxxxxxx>
---
 drivers/usb/host/xhci-plat.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index b0c8e8efc43b..6a45ee431945 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -294,10 +294,6 @@ int xhci_plat_probe(struct platform_device *pdev, struct device *sysdev, const s
 		xhci->shared_hcd->usb_phy = devm_usb_get_phy_by_phandle(sysdev,
 			    "usb-phy", 1);
 		if (IS_ERR(xhci->shared_hcd->usb_phy)) {
-			if (PTR_ERR(xhci->shared_hcd->usb_phy) != -ENODEV)
-				dev_err(sysdev, "%s get usb3phy fail (ret=%d)\n",
-					     __func__,
-					    (int)PTR_ERR(xhci->shared_hcd->usb_phy));
 			xhci->shared_hcd->usb_phy = NULL;
 		} else {
 			ret = usb_phy_init(xhci->shared_hcd->usb_phy);
-- 
2.34.1
